dbt_style_guide.md

```diff
@@ -6,87 +6,148 @@ Our models (typically) fit into three main categories: staging, marts, base/inte
+  - If switching to `incremental`, this should be specified in the model's configuration.
 
 ## dbt conventions
-* Only `stg_` models (or `base_` models if your project requires them) should select from `source`s.
-* All other models should only select from other models.
+- Only models in the `staging` folder should select from [sources](https://docs.getdbt.com/docs/building-a-dbt-project/using-sources)
+- Models not in the `staging` folder should select from [refs](https://docs.getdbt.com/reference/dbt-jinja-functions/ref).
 
 ## Testing
 
-- Every subdirectory should contain a `.yml` file, in which each model in the subdirectory is tested. For staging folders, the naming structure should be `src_sourcename.yml`. For other folders, the structure should be `foldername.yml` (example `core.yml`).
-- At a minimum, unique and not_null tests should be applied to the primary key of each model.
+- At a minimum, unique and not_null tests should be applied to the assumed primary key of each model.
 
 ## Naming and field conventions
 
-* Schema, table and column names should be in `snake_case`.
-* Use names based on the _business_ terminology, rather than the source terminology.
-* Each model should have a primary key.
-* The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
-* For base/staging models, fields should be ordered in categories, where identifiers are first and timestamps are at the end.
-* Timestamp columns should be named `<event>_at`, e.g. `created_at`, and should be in UTC. If a different timezone is being used, this should be indicated with a suffix, e.g `created_at_pt`.
-* Booleans should be prefixed with `is_` or `has_`.
-* Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
-* Avoid reserved words as column names
-* Consistency is key! Use the same field names across models where possible, e.g. a key to the `customers` table should be named `customer_id` rather than `user_id`.
+- Schema, table and column names should be in `snake_case`.
+
+- Use names based on the _business_ terminology, rather than the source terminology.
+
+- Each model should have a primary key that can identify the unique row.
+
+- The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
+
+- If a surrogate key is created, it should be named `<object>_sk`.
+
+- For `base` or `staging` models, columns should be ordered in categories, where identifiers are first and date/time fields are at the end.  
+  Example:
+  ```sql
+  transformed as (
+
+      select
+
+        -- ids
+        order_id,
+        customer_id,
+
+        -- dimensions

```

Comment:
should this be `varchars` to match numeric/boolean/datetime?

Reply:
> columns should be ordered in categories, where identifiers are first and date/time fields are at the end.

It might be productive to think of this less in terms of data type and more in terms of function. In this example, I see four functions:
1. Identifiers - Primary and foreign keys.
2. Dimensions - Qualitative values used to categorize or segment. A boolean value in this case counts as qualitative.
3. Measures - Quantitative values that you can measure, like the total value of an order
4. Metadata - Data about the data. This would be timestamps or other internal fields.

Using this framework, we get nearly the same order, but with a cleaner abstraction:

```sql
 transformed as (
      select

          -- ids
          order_id,
          customer_id,

          -- dimensions
          order_status,
          is_shipped,

          -- measures
          order_total,
        
          -- metadata
          created_at,
          updated_at,
          _sdc_batched_at

      from source
  )
```
Food for thoughts 🤷

Reply:
👍  Nicholas's idea seems really straightforward

Reply:
^ agree

Reply:
+1 to @nicholasyager!

Reply:
Changed this ([commit](https://github.com/dbt-labs/corp/pull/63/commits/47ca66a46843097175638c0c73c845d49059013f)) to regroup by Nicholas' categories.